frontend/src/app/services/http.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable, throwError } from 'rxjs';
+
+const url = 'http://localhost:8080';
+const loginEndpoint = '/user/login';
+const minLength = 3;
+
+export interface LoginResponse {
+  token: string;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class HttpService {
```

Comment:
I do recommend a more specific name for this service. `UserService` perhaps?

Reply:
Right now this service does one job and it is simple http request, maybe in the future, when there will be a few request types and business logic will become more complex, I would think about splitting them logically.